go-controller/pkg/ovn/egressgw.go

```diff
@@ -47,7 +47,9 @@ type ovnRoute struct {
 }
 
 type externalRouteInfo struct {
-	sync.RWMutex
+	sync.Mutex
+	deleted bool
```

Comment:
We call `deleteRouteInfoLocked` only from `deleteGWRoutesForPod` which is only called from `deleteLogicalPort`....That is the only path where the pod is entirely removed from the cache. We are sure that we only need to care about the `did it get deleted from the cache` scenario right? I guess we won'tbe in a `did it get re-added` scenario? Since the pod would actually have to be recreated of sorts and since ns/name is unique we should be fine,,

Reply:
Our pod events for the same pod are serialized so that greatly lessens this kind of race. But as above, there is a slight chance that a quick create/delete/create could fail the second create, because the `delete` goroutine hasn't run yet and the pod gets added again, and the error is returned.

But, and I just realized this, that makes our pod handler retry! So we're good here anyway, because even if it fails to be updated because it's marked deleted, the pod add handler will retry a bit later and by that point the goroutine will have run, and we'll be OK.

Reply:
oh yes we will retry! good point. sg